### PR TITLE
[DO NOT MERGE] Promise/A+ tests through 2.2.3

### PR DESCRIPTION
**Pull request by CaptEmulation**
Initial round of tests as I head out to lunch, for your review.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ module.exports = function (grunt) {
   grunt.loadNpmTasks('grunt-mocha-test');
   grunt.loadNpmTasks('grunt-contrib-uglify');
 
-  grunt.registerTask('test', ['jscs', 'jshint', 'uglify', 'mochaTest']);
+  grunt.registerTask('test', ['uglify', 'mochaTest', 'jscs', 'jshint']);
```
Comment (neonphog) [inline Gruntfile.js]: could be:

```javascript
grunt.registerTask('test', ['uglify', 'mochaTest']);
grunt.registerTask('lint', ['jscs', 'jshint']);
grunt.registerTask('default', ['lint', 'test']);
```

File: test/test.js [hunk 2]
```diff
@@ -343,6 +392,314 @@
...
+            describe('2.2.3.1 it must be called after promise is rejected, ' +
+              'with promise’s reason as its first argument.', function () {
+              it('before being resolved', function (done) {
+                var p = lib.promise();
+                var value = {};
+                var onRejected = sinon.stub();
+
+                p.then(onRejected);
+
+                p.resolve(value);
```
Comment (neonphog) [inline test/test.js]:
```javascript
p.then(null, onRejected);
p.reject(value);
```

Comment (neonphog) [inline test/test.js]: Ah, I see, this whole section is a copy of the onFulfilled section.

**Comment by neonphog** [comment]
Bwahaha, I'm strangely excited for these, looking good.

Also, if you wish, you should add yourself to the package.json contributors.

File: Gruntfile.js (hunk 1 above)
Comment (CaptEmulation) [inline Gruntfile.js]: ok

**Comment by CaptEmulation** [comment]
2.2.6 is a failing test

**Pull request merged**